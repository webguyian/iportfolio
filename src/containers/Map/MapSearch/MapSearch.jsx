import React from 'react';
import PropTypes from 'prop-types';

import Button from 'components/Button/Button';

const MapSearch = props => {
  const { inputRef } = props;
  const baseClass = 'map-search';
  const handleSubmit = event => event.preventDefault();

  return (
    <form className={baseClass} onSubmit={handleSubmit}>
      <input
        className={`${baseClass}-input`}
        placeholder="Search here"
        ref={inputRef}
      />
      <Button className={`${baseClass}-btn`} icon="search" type="submit">
        Search
      </Button>
    </form>
  );
};

MapSearch.propTypes = {
  inputRef: PropTypes.object
};

export default MapSearch;
