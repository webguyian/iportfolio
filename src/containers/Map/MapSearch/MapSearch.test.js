import React from 'react';
import { act, create } from 'react-test-renderer';

import MapSearch from './MapSearch';

describe('<MapSearch />', () => {
  const props = {
    inputRef: React.createRef()
  };

  it('renders correctly', () => {
    const component = create(<MapSearch {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('handles submit event', () => {
    const component = create(<MapSearch {...props} />);
    const form = component.root.findByType('form');
    const mockEvent = { preventDefault: jest.fn() };

    expect(mockEvent.preventDefault).not.toHaveBeenCalled();
    act(() => {
      form.props.onSubmit(mockEvent);
    });
    expect(mockEvent.preventDefault).toHaveBeenCalled();
  });
});
