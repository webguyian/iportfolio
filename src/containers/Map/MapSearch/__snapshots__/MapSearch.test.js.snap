// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapSearch /> renders correctly 1`] = `
<form
  className="map-search"
  onSubmit={[Function]}
>
  <input
    className="map-search-input"
    placeholder="Search here"
  />
  <button
    className="ui-btn ui-btn--with-icon map-search-btn"
    type="submit"
  >
    <i
      className="ui-icon ui-icon-search fas fa-search"
    />
    <span
      className="ui-text ui-text--accessible"
    >
      Search
    </span>
  </button>
</form>
`;
